proto/regen/data/v2/tx.proto

```diff
@@ -106,20 +106,39 @@ message MsgAttestResponse {
 
 // MsgDefineResolver is the Msg/DefineResolver request type.
 message MsgDefineResolver {
-  option (cosmos.msg.v1.signer) = "manager";
+  option (cosmos.msg.v1.signer) = "definer";
 
-  // manager is the address of the resolver manager. The manager is able
-  // to make future calls using the ID returned by this operation with
-  // Msg/RegisterResolver. To authorize other accounts to register resolvers,
-  // the manager should make use of cosmos.authz.
-  string manager = 1;
+  // definer is the address of the account defining the resolver. If
+  // the boolean public is set to true, then any user can register
+  // data with this resolver. If the boolean public is set to false,
+  // then only the definer can register data with this resolver and
+  // must use a feature such as cosmos.authz to authorize other accounts
+  // to register data with this resolver.
+  string definer = 1;
 
-  // resolver_url is a resolver URL which should refer to an HTTP service
-  // which will respond to a GET request with the IRI of a ContentHash
+  // resolver_url is a resolver URL.
+  //
+  // If it refers to an HTTP URL, that HTTP service should
+  // respond to a GET request with the IRI of a ContentHash as the path parameter
   // and return the content if it exists or a 404. For graph data, resolvers
   // should use the HTTP Accept header to negotiate the RDF serialization
   // format.
+  //
+  // To use IPFS, the resolver_url ipfs: should be defined with public set to true
+  // and used as the resolver for any data hosted on IPFS. Content hashes must be
+  // adapted to IPFS's CID format. The multicodec raw (0x55) should
```

Comment:
> Content hashes must be adapted to IPFS's CID format.

Just want to clarify how this works.. For both raw and graph content hashes, the `hash` will be set to the full `cid`, where the `cid` was generated using either the multicodec raw (0x55) or multicodec rdfc-1 (0xb403)?

Reply:
The content hash on Regen would still be the same as if it were registered w/ an HTTP resolver. I think this comment just communicates that users must take the resulting Regen IRI, and convert it to a specific IPFS CID in order to fetch the data from the IPFS network.

Reply:
Exactly

Reply:
Ah thanks, this makes more sense. With an example I think this would be very clear. Hopefully soon...